### DISCUSSION TITLE
Extended Menu (the "shell" component)

### DISCUSSION THREAD

**Discussion by pchemguy** [sqlpage/SQLPage#429] [Show and tell]
I have been working on extending the "shell" component to provided extra menu features. I need to finish the docs before submitting the updated "shell" component, but I wanted to share demo screenshots in advance.

Basically, I wanted to have icons in menus and button-styled menu items, as demonstrated on screenshots. As illustrated, menus, which use the present definition, are not affected. The extra options permit adding icons to menu elements. as well as turning both top-level and submenu items into buttons. I also added a flag which can be used to remove the extra white spacing on the right of submenus (the last screenshot). I welcome any feedback on this.

---

![extended_menu_demo_1](https://github.com/lovasoa/SQLpage/assets/39730837/c41048fa-92f5-426e-b39b-f23c22f19b56)

---

![extended_menu_demo_2](https://github.com/lovasoa/SQLpage/assets/39730837/8343ad6a-5eae-4579-bda2-f014f1279d23)

---

![extended_menu_demo_3](https://github.com/lovasoa/SQLpage/assets/39730837/959b2907-94c9-4832-a616-9f90befbec9f)

**Comment by lovasoa** [comment]
The new options sound useful ! But I feel like there is a problem with padding and margins in the design...

**Reply by pchemguy** [reply]
I suppose, horizontal spacing between top-level button should be adjusted. That what you mean?

**Reply by lovasoa** [reply]
Yes, the icon buttons in the menu look cramped. I don't know what exactly should be done, but I think it doesn't feel professionally designed.

The meny that says tabs, layouts, forms also has a huge right margin.

**Reply by pchemguy** [reply]
When you place both conventional items and buttons, the spacing can be reduced but removing it completely requires additional tweaking. But check "Community" menu on the website. That is the white space I can remove. When submenu contains buttons only, all elements have the same core class (btn), so it is simpler.

**Reply by lovasoa** [reply]
I think there isn't anything visually shocking here
![image](https://github.com/lovasoa/SQLpage/assets/552629/f676eec0-eb19-4993-835e-c4e8d77c9f46)

But I think there is a spacing problem here

![image](https://github.com/lovasoa/SQLpage/assets/552629/a4bfdd57-58cd-4f98-8738-c2ce8d44ae14)

and here

![image](https://github.com/lovasoa/SQLpage/assets/552629/59dd3ac5-5fcb-433c-b181-2068b9d0a408)

**Reply by pchemguy** [reply]
Right, thanks

**Comment by pchemguy** [comment]
I have made some progress. It is not perfect but looks better. I have created a pull request, which also includes a demo menu file. Please take a look and let me know what you think.
